subsys/bluetooth/audio/gmap_client.c

```diff
@@ -648,7 +655,7 @@ int bt_gmap_discover(struct bt_conn *conn)
 
 	gmap_cli = &gmap_insts[bt_conn_index(conn)];
 
-	if (gmap_cli->busy) {
+	if (atomic_test_and_set_bit(gmap_cli->flags, GMAP_CLIENT_FLAG_BUSY)) {
```

Comment:
Where was `gmap_cli->busy` getting set to `true` previously? I can't see you modify that place in this PR. You're adding the analogue to that here (`set_bit()`), but seems like the original `bool` was never actually set anywhere?

Reply:
Looks like you are correct. It was, incorrectly, never set to `true` it seems. 

Arguably we should add BSIM tests for this behavior as well